src/coreclr/vm/arm64/asmhelpers.S

```diff
@@ -322,44 +349,13 @@ WRITE_BARRIER_ENTRY JIT_CheckedWriteBarrier
     // branch below is not taken.
```

Comment:
The lines `ldr  x12,  LOCAL_LABEL(wbs_lowest_address)` and `ldr  x12, LOCAL_LABEL(wbs_highest_address)` above were referencing the unmodified version of `JIT_WriteBarrier_Table`.

By moving `JIT_CheckedWriteBarrier` into `JIT_PatchedCodeStart` the relative address are correct.

`JIT_ByRefWriteBarrier` calls `JIT_CheckedWriteBarrier`, so it is simplest to just move it too.  It also removes a layer of indirection which would need to be added.